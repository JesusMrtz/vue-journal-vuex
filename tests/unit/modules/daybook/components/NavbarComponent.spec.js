import { shallowMount } from "@vue/test-utils"
import NavbarComponent from '@/modules/daybook/components/NavbarComponent'
import createVuexStore from "../../../mocks/mock-store"


const mockRouter = {
  push: jest.fn()
}

jest.mock('vue-router', () => ({
  useRouter: () => mockRouter
}))

describe('Pruebas en el Navbar component', () => {
  const store = createVuexStore({
    user: {
      name: 'Jesús',
      email: 'correo@correo.co'
    },
    status: 'authenticated',
    idToken: 'ABC',
    refreshToken: '123'
  })

  beforeEach(() => {
    jest.clearAllMocks()
  })

  
  it('Debe de mostrar el componente correctamente', () => {
    const wrapper = shallowMount(NavbarComponent, {
      global: {
        plugins: [ store ]
      }
    })

    expect(wrapper.html()).toMatchSnapshot()
  })


  it('Click en el logout, debe de cerrar sesión y redireccionar', async () => {
    const wrapper = shallowMount(NavbarComponent, {
      global: {
        plugins: [ store ]
      }
    })

    await wrapper.find('button').trigger('click')

    expect(mockRouter.push).toHaveBeenCalledTimes(1)
    expect(mockRouter.push).toHaveBeenCalledWith({ name: 'login' })
    expect(store.state.auth).toEqual({
      user: null,
      status: 'not-authenticated',
      idToken: null,
      refreshToken: null
    })
  })
})