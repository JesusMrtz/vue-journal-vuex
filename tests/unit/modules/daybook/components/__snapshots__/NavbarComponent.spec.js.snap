// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el Navbar component Debe de mostrar el componente correctamente 1`] = `
<nav class="navbar bg-primary"><a class="navbar-brand text-white" href="#"><img class="d-inline-block align-text-top mx-2" src="" alt="Vue Logo" height="24"> Jes√∫s</a>
  <div class="d-flex"><button type="button" class="btn btn-outline-info mx-2"><i class="fa fa-sign-out-alt"></i></button></div>
</nav>
`;
